src/Rings/mpoly-ideals.jl

```diff
@@ -112,9 +112,11 @@ ideal(x, y, z^2)
 ```
 """
 function Base.:+(I::MPolyIdeal{T}, J::MPolyIdeal{T}) where T
-  singular_assure(I)
-  singular_assure(J)
-  return MPolyIdeal(base_ring(I), I.gens.S + J.gens.S)
+  if base_ring(I) == base_ring(J)
+    return MPolyIdeal(unique(vcat(I.gens.O,J.gens.O)))
+  else
+    error("Not possible due to different base rings.")
+  end
```

Comment:
can you make a helper function `check_base_rings(I::MPolyIdeal{T}, J::MPolyIdeal{T}))` so this code can be written more cleanly as the following. This will help in the other functions that also require base ring checks.
```
  check_base_rings(I, J)
  return MPolyIdeal(unique(vcat(I.gens.O,J.gens.O)))
```

Reply:
Will do so once I fixed the other problems.